main.tf

```diff
@@ -93,7 +93,7 @@ resource "ibm_is_virtual_endpoint_gateway" "vpe" {
   target {
     name          = length(regexall("crn:v1:([^:]*:){6}", each.value.crn)) > 0 ? null : each.value.crn
     crn           = length(regexall("crn:v1:([^:]*:){6}", each.value.crn)) > 0 ? each.value.crn : null
-    resource_type = length(regexall("crn:v1:([^:]*:){6}", each.value.crn)) > 0 ? "provider_cloud_service" : "provider_infrastructure_service"
+    resource_type = length(regexall("crn:v1:([^:]*:){6}", each.value.crn)) > 0 ? can(regex(".*private-path-service-gateway.*", each.value.crn)) ? "private_path_service_gateway" : "provider_cloud_service" : "provider_infrastructure_service"
```

Comment:
Your `can(regex())` here will work, but since your regex is so simple (you are just looking for a simple string in the crn) you can accomplish the same functionality I think with a simple `strcontains()`: https://developer.hashicorp.com/terraform/language/functions/strcontains